### PR TITLE
Source Github: add 3 new streams (workflows, workflow runs and repository vulnerability)

### PR DESCRIPTION
**Pull request by bzAmin**
…eam for the connector asana

## What
*Describe what the change is solving*
For 2 connectors GitHub and asana
GitHub : Adding 3 streams Workflows, Workflow runs and vulnerability alert
Asana : We collect all the stories of a task but we don't have the task id. Added in the parse_response.

## How
Create 3 new streams for the GitHub source.
Add a parse response for the Asana stream "stories" to add the task_id

## Recommended reading order
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.
Not impactful for the users

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/airbytehq/airbyte/9251)

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@bzAmin can you sign the CLA?

**Comment by marcosmarxm** [comment]
@bzAmin can you run `./gradlew format` and bump the connectors version for Asana and Github? see [here](https://docs.airbyte.io/connector-development#4.-publish-the-connector)

File: airbyte-integrations/connectors/source-github/source_github/streams.py [hunk 1]
```diff
@@ -802,3 +802,67 @@ class PullRequestCommentReactions(ReactionStream):
...
+    Get the dependabot alerts of a GitHub repository
+    API documentation: https://docs.github.com/en/graphql/reference/objects#repositoryvulnerabilityalert
+    """ 
+     
+    @property
+    def http_method(self) -> str:
+        return "POST"
+    
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"graphql"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-github/source_github/streams.py:852]: remove the f-string.

File: airbyte-integrations/connectors/source-github/source_github/streams.py [hunk 2]
```diff
@@ -802,3 +802,67 @@ class PullRequestCommentReactions(ReactionStream):
...
+    
+    def parse_response(self, response: requests.Response, stream_slice: Mapping[str, Any] = None, **kwargs) -> Iterable[Mapping]:
+        response = response.json()
+        response["repository"] = stream_slice['repository']
+        yield response
+
+
+# Security streams
+
+class Security(GithubStream):
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-github/source_github/streams.py:841]: Maybe a more meaningful name like `RepositoryVulnerabilityAlert`

**Comment by bzAmin** [comment]
Hello ! I performed the changes. I'm waiting the review to perform the third step in the documentation publishing a connector : 

One of Airbyte maintainers will review the change and publish the new version of the connector to Docker hub. Triggering tests and publishing connectors can be done by leaving a comment on the PR with the following format (the PR must be from the Airbyte repo, not a fork):

**Comment by bzAmin** [comment]
concerning the ./gradlew format : I ran ./gradlew :airbyte-integrations:connectors:source-github:build. Is it what you meant? 
Thanks for your review. 

image

**Comment by marcosmarxm** [comment]
> concerning the ./gradlew format : I ran ./gradlew :airbyte-integrations:connectors:source-github:build. Is it what you meant? Thanks for your review.
> 
> image

yes, thanks @bzAmin 
also can you sign the CLA?
image

**Comment by bzAmin** [comment]
Hello ! Done

**Comment by bzAmin** [comment]
Hello, any news on this pull request ? thanks for your review.

File: airbyte-integrations/connectors/source-github/source_github/streams.py [hunk 3]
```diff
@@ -802,3 +802,75 @@ class PullRequestCommentReactions(ReactionStream):
...
+
+    def request_body_json(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        body = (
+            '{ repository(name: "'
+            + stream_slice["repository"].split("/")[1]
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-github/source_github/streams.py:865]: @bzAmin can you give a variable name for what the index [0] and [1] are? like:
```python
repo = stream_slice["repository"].split("/")[1]
org... [0]?
```
The goal is to make this clear to anyone looking into this code.

File: airbyte-integrations/connectors/source-asana/source_asana/streams.py [hunk 4]
```diff
@@ -161,6 +161,12 @@ def stream_slices(
     ) -> Iterable[Optional[Mapping[str, Any]]]:
         yield from self.read_slices_from_records(stream_class=Tasks, slice_field="task_gid")
 
+    def parse_response(self, response: requests.Response, stream_slice: Mapping[str, Any] = None,**kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        records = response_json.get("data", []) # Asana puts records in a container array "data"
+        for record in records:
+            record["task_gid"] = stream_slice["task_gid"]
+        yield from records
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-asana/source_asana/streams.py:169]: can I suggest to you split the changes made in Asana in another PR?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
some comments, sorry the delay @bzAmin !!!

**Comment by bzAmin** [comment]
Hello @marcosmarxm changes has been made. Thanks for your review

File: airbyte-integrations/connectors/source-github/source_github/streams.py (hunk 3 above)
Comment (bzAmin) [reply airbyte-integrations/connectors/source-github/source_github/streams.py]: Both has been put in variables

File: airbyte-integrations/connectors/source-asana/source_asana/streams.py (hunk 4 above)
Comment (bzAmin) [reply airbyte-integrations/connectors/source-asana/source_asana/streams.py]: Changes on the asana connector have been removed

**Review by marcosmarxm — APPROVED** [review]

**Comment by bzAmin** [comment]
Hello @marcosmarxm, thanks for your code approval. Can you please approuve the workflow?

**Comment by marcosmarxm** [comment]
sorry the delay I need to remove the vulnerability report, feel free to submit another PR adding that stream.

**Pull request merged**